hamocc/mo_carchm.F90

```diff
@@ -901,17 +906,19 @@ subroutine carchm_solve(saln,tc,ta,sit,pt,K1,K2,Kb,Kw,Ks1,Kf,Ksi,K1p,K2p,K3p,ah1
       ah2o = sqrt( ( tc - ac )**2 + 4. * ( ac * K2 / K1 ) * ( 2. * tc - ac ) )
       ah2  = 0.5 * K1 / ac *( ( tc - ac ) + ah2o )
       erel = ( ah2 - ah1 ) / ah2
-      if (abs( erel ).ge.eps) then
-        ah1 = ah2
+      if (abs( erel ) >= eps) then
+        ! make sure [H+] stays between pH 5 and 11
+        ah1 = max(1.0e-11,min(ah2,1.0e-5))
```

Comment:
```suggestion
        ah1 = max(min_ah,min(ah2,max_ah))
```
Maybe introduce this as (private) static parameters in mo_carchm? Reduce risk of divergence between carchm_solve and carchm_solve_dicsat.

Reply:
Yes, this would make sense, will change accordingly